OWML.ModHelper.Menus/ModPromptButton.cs

```diff
@@ -0,0 +1,89 @@
+﻿using OWML.Common.Menus;
+using OWML.ModHelper.Events;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace OWML.ModHelper.Menus
+{
+    public class ModPromptButton : ModButton, IModPromptButton
+    {
+        private ScreenPrompt _prompt;
+        private UITextType _textID;
+        private readonly ButtonWithHotkeyImageElement _hotkeyButton;
+
+        public string DefaultTitle
+        {
+            get => UITextLibrary.GetString(_textID);
+        }
+        public ScreenPrompt Prompt
+        {
+            get => _prompt;
+            set
+            {
+                _prompt = value;
+                _hotkeyButton.SetPrompt(value);
+            }
+        }
+
+        public ModPromptButton(Button button, IModMenu menu) : base(button, menu)
+        {
+            _hotkeyButton = Button.GetComponent<ButtonWithHotkeyImageElement>();
+            _textID = Button.GetComponentInChildren<LocalizedText>(true).GetValue<UITextType>("_textID");
```

Comment:
This doesn't have to happen if the null check below fails, so move it below the null check?

Reply:
K